src/west/app/project.py

```diff
@@ -845,6 +855,77 @@ def handle(group_filter_item):
 
         self.fs = self.fetch_strategy()
 
+    @staticmethod
+    def _update_concurrent_thread(args):
+        project, self, logdir = args
+        if logdir:
+            stdout = os.dup(sys.stdout.fileno())
+            stderr = os.dup(sys.stderr.fileno())
+
+            logfile = open(os.path.join(logdir, project.name), "wb")
+            os.dup2(logfile.fileno(), sys.stdout.fileno())
+            os.dup2(logfile.fileno(), sys.stderr.fileno())
+
+        try:
+            try:
+                self.update(project)
+                return True
+            except subprocess.CalledProcessError:
+                return False
+        finally:
+            if logdir:
+                sys.stdout.flush()
+                sys.stderr.flush()
+
+                os.dup2(stdout, sys.stdout.fileno())
+                os.dup2(stderr, sys.stderr.fileno())
+
+    def update_project_list(self, projects):
+        failed = []
+        updated = []
+
+        if self.args.jobs > 1:
+            # XXX: this is not picklable
+            self.parser = None
+
+            if self.args.unbuffered:
+                logdir = None
+                logdir_name = None
+            else:
+                logdir = tempfile.TemporaryDirectory()
+                logdir_name = logdir.name
+
+            args = []
+            for project in projects:
+                args.append((project, self, logdir_name))
+
+            with multiprocessing.Pool(self.args.jobs) as p:
+                for index, successful in enumerate(
+                        p.imap_unordered(self._update_concurrent_thread,
+                                         args)):
+                    project = projects[index]
+
+                    if logdir:
+                        logfile = open(os.path.join(logdir_name,
+                                       project.name), "rb")
+                        sys.stderr.buffer.write(logfile.read())
+                        sys.stderr.flush()
```

Comment:
It seems weird and very unusual to redirect stdout to stderr and it feels like this could be a cause for false alarms. Please add a comment explaining why you do that, probably because git (strangely?) prints a lot of harmless  info to stderr and ( if you know why git does that then even better: say why!).